sdk/storage/azure-storage-common/src/storage_retry_policy.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/storage/common/storage_retry_policy.hpp"
+#include "azure/storage/common/storage_common.hpp"
+
+#include <thread>
+
+namespace Azure { namespace Storage {
+
+  std::unique_ptr<Azure::Core::Http::RawResponse> StorageRetryPolicy::Send(
+      const Azure::Core::Context& ctx,
+      Azure::Core::Http::Request& request,
+      Azure::Core::Http::NextHttpPolicy nextHttpPolicy) const
+  {
+    bool considerSecondary = (request.GetMethod() == Azure::Core::Http::HttpMethod::Get
+                              || request.GetMethod() == Azure::Core::Http::HttpMethod::Head)
+        && !m_secondaryHost.empty();
+
+    std::string primaryHost = request.GetUrl().GetHost();
+    const std::string& secondaryHost = m_secondaryHost;
+
+    bool isUsingSecondary = false;
+
+    auto switchHost
```

Comment:
Re-sending the same request more than one time could end up sending the request to secondaryHost only???

Say I create my request and send it.  It fails and switches to m_secondaryHost. When this happens, the original host gets replaced by the secondary.  
Then, if secondary works fine and request returns, the Host is still using secondary.
Then every time this request is reused will never try to reach primary again.

This is fine if there is no way a request can be re-used.

Reply:
`Request` struct is never reused